rust-tdshim/src/acpi.rs

```diff
@@ -135,8 +138,36 @@ impl AcpiTables {
     pub fn finish(&mut self) -> u64 {
         // Create XSDT
         let mut xsdt = Xsdt::new();
-        for i in 0..self.table_num {
-            xsdt.add_table(self.table_addr[i]);
+
+        // The Fixed ACPI Description Table (FADT) should
+        // always be the first table that XSDT points to.
+        //
+        if let Some(fadt) = self.fadt {
+            let fadt_header = GenericSdtHeader::read_from(
+                &self.acpi_memory[fadt..fadt + size_of::<GenericSdtHeader>()],
+            )
+            .unwrap();
+            if fadt_header.revision == ACPI_FADT_MAJOR_VERSION {
```

Comment:
Why the code checks FADT version?
DSDT should be included in FADT in any version.

Reply:
Agree, it is a mistake. I have updated the patch. DSDT table will be linked to DSDT field(4-bytes) of FADT since ACPI memory of td-layout is below 4G.